### PR TITLE
BugFix: Incorrect conversion between integer types

### PR DESCRIPTION
**Pull request by rpl-ffl**
Incorrect conversion between integer types
https://codeql.github.com/codeql-query-help/go/go-incorrect-integer-conversion/

Fixes https://github.com/Fantom-foundation/Aida/issues/1109

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Comment by rpl-ffl** [comment]
https://github.com/Fantom-foundation/Aida/security/code-scanning?query=is%3Aopen+pr%3A1112
The 2 warnings have now disappeared.

**Review by petr-hanzl — APPROVED** [review]

**Comment by rpl-ffl** [comment]
@wsodsong: `int` -> `uint64` is wrong since we want the entire pipeline to be `int`. According changes incoming.

File: cmd/util-db/db/metadata.go [hunk 1]
```diff
@@ -174,7 +174,7 @@ func insertMetadata(ctx *cli.Context) error {
 			return err
 		}
 	case utils.TypePrefix:
-		num, err := strconv.Atoi(valArg)
+		num, err := strconv.ParseUint(valArg, 10, 8)
```
Comment (petr-hanzl) [inline cmd/util-db/db/metadata.go:177]: Think about using `strconv.ParseInt()` and using int64 instead of parsing Uint64 into Int32.

Comment (rpl-ffl) [reply cmd/util-db/db/metadata.go:177]: Not really sure if you're marking the right place?
We are trying to get a byte from string, so I'm parsing with Uint8.

Comment (rpl-ffl) [reply cmd/util-db/db/metadata.go:177]: I see your point. The underlying type of the return int is always uint64 when using `ParseUInt`.

Comment (rpl-ffl) [reply cmd/util-db/db/metadata.go]: I think I see your point. The result of `strconv.ParseUInt()` is Uint64, so we should convert this to Uint8 before giving it to byte.

**Pull request merged**